nucypher/blockchain/eth/actors.py

```diff
@@ -1194,15 +1198,80 @@ def __init__(self,
 
         # Workers cannot be started without being assigned a stake first.
         if is_me:
+            if check_active_worker:
+                self._checksum_address = self.block_until_ready(worker_address, self.registry)
             self.stakes = StakeList(registry=self.registry, checksum_address=self.checksum_address)
             self.stakes.refresh()
-            if check_active_worker and not len(self.stakes):
-                raise self.DetachedWorker(f"{self.__worker_address} is not bonded to {self.checksum_address}.")
 
             self.work_tracker = work_tracker or WorkTracker(worker=self)
             if start_working_now:
                 self.work_tracker.start(act_now=False)
 
+    @classmethod
+    def block_until_ready(cls,
+                          worker_address: str,
+                          registry: 'BaseContractRegistry',
+                          poll_rate: int = None,
+                          timeout: int = None) -> str:
```

Comment:
Could this be an instance method? Because from first glance this make more sense for instance than class. Maybe I'm missing something

Reply:
:heavy_check_mark: